[kodi] NPE after receiving "OnPlay" event from server

## Current behavior

log in openHAB 2.1.0 -  snapshot 1020 -
```
2017-08-28 22:11:28.089 [DEBUG] [i.internal.protocol.KodiClientSocket] - Message received from server: {"jsonrpc":"2.0","method":"Player.OnPlay","params":{"data":{"item":{"title":"Dagsrevyen 21","type":"movie"},"player":{"playerid":-1,"speed":1}},"sender":"xbmc"}}
2017-08-28 22:11:28.089 [DEBUG] [i.internal.protocol.KodiClientSocket] - Event received from server: {"jsonrpc":"2.0","method":"Player.OnPlay","params":{"data":{"item":{"title":"Dagsrevyen 21","type":"movie"},"player":{"playerid":-1,"speed":1}},"sender":"xbmc"}}
2017-08-28 22:11:28.099 [DEBUG] [i.internal.protocol.KodiClientSocket] - send message: {"jsonrpc":"2.0","id":3009,"method":"Player.GetItem","params":{"playerid":-1,"properties":["title","album","artist","director","thumbnail","file","fanart","showtitle","streamdetails","channel","channeltype"]}}
2017-08-28 22:11:28.100 [DEBUG] [i.internal.protocol.KodiClientSocket] - Message received from server: {"error":{"code":-32602,"data":{"method":"Player.GetItem","stack":{"message":"Value between 0 (inclusive) and 2 (inclusive) expected but -1 received","name":"playerid","type":"integer"}},"message":"Invalid params."},"id":3009,"jsonrpc":"2.0"}
2017-08-28 22:11:28.101 [DEBUG] [i.internal.protocol.KodiClientSocket] - Response received from server: {"error":{"code":-32602,"data":{"method":"Player.GetItem","stack":{"message":"Value between 0 (inclusive) and 2 (inclusive) expected but -1 received","name":"playerid","type":"integer"}},"message":"Invalid params."},"id":3009,"jsonrpc":"2.0"}
2017-08-28 22:11:28.102 [DEBUG] [i.internal.protocol.KodiClientSocket] - callMethod returns {"error":{"code":-32602,"data":{"method":"Player.GetItem","stack":{"message":"Value between 0 (inclusive) and 2 (inclusive) expected but -1 received","name":"playerid","type":"integer"}},"message":"Invalid params."},"id":3009,"jsonrpc":"2.0"}
2017-08-28 22:11:28.104 [INFO ] [.eclipse.smarthome.model.script.Kodi] - KodiStopped.state=OFF
2017-08-28 22:11:28.106 [ERROR] [i.internal.protocol.KodiClientSocket] - Error handling event {"jsonrpc":"2.0","method":"Player.OnPlay","params":{"data":{"item":{"title":"Dagsrevyen 21","type":"movie"},"player":{"playerid":-1,"speed":1}},"sender":"xbmc"}} player state change message: null
java.lang.NullPointerException: null
        at org.openhab.binding.kodi.internal.protocol.KodiConnection.requestPlayerUpdate(KodiConnection.java:234) [184:org.openhab.binding.kodi:2.2.0.201708201100]
        at org.openhab.binding.kodi.internal.protocol.KodiConnection.processPlayerStateChanged(KodiConnection.java:377) [184:org.openhab.binding.kodi:2.2.0.201708201100]
        at org.openhab.binding.kodi.internal.protocol.KodiConnection.handleEvent(KodiConnection.java:354) [184:org.openhab.binding.kodi:2.2.0.201708201100]
        at org.openhab.binding.kodi.internal.protocol.KodiClientSocket$KodiWebSocketListener$2.run(KodiClientSocket.java:158) [184:org.openhab.binding.kodi:2.2.0.201708201100]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```

## Related sources
Related discussion in community: https://community.openhab.org/t/kodi-binding-exception-when-starting-pvr-recording/33674

--- a/addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java
+++ b/addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java
@@ -222,14 +222,20 @@ public class KodiClientSocket {
 
             sendMessage(message);
             if (commandLatch.await(REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
-                logger.debug("callMethod returns {}", commandResponse.toString());
-                return commandResponse.get("result");
+                logger.debug("callMethod returns {}", commandResponse);
+                if (commandResponse.has("result")) {
+                    return commandResponse.get("result");
+                } else {
+                    JsonElement error = commandResponse.get("error");
+                    logger.debug("Error received from server: {}", error != null ? error : "");
+                    return null;
+                }
             } else {
-                logger.error("Timeout during callMethod({}, {})", methodName, params != null ? params.toString() : "");
+                logger.debug("Timeout during callMethod({}, {})", methodName, params != null ? params : "");
                 return null;
             }
         } catch (Exception e) {
-            logger.error("Error during callMethod", e);
+            logger.debug("Error during callMethod({}): {}", methodName, e.getMessage(), e);
             return null;
         }
     }

--- a/addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java
+++ b/addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java
@@ -227,11 +227,11 @@ public class KodiClientSocket {
                     return commandResponse.get("result");
                 } else {
                     JsonElement error = commandResponse.get("error");
-                    logger.debug("Error received from server: {}", error != null ? error : "");
+                    logger.debug("Error received from server: {}", error);
                     return null;
                 }
             } else {
-                logger.debug("Timeout during callMethod({}, {})", methodName, params != null ? params : "");
+                logger.debug("Timeout during callMethod({}, {})", methodName, params);
                 return null;
             }
         } catch (Exception e) {

--- a/addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java
+++ b/addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java
@@ -69,11 +69,14 @@ public class KodiConnection implements KodiClientSocketEventListener {
 
     private int getActivePlayer() {
         JsonElement response = socket.callMethod("Player.GetActivePlayers");
-        if (response != null) {
-            boolean playing = response.isJsonArray() && response.getAsJsonArray().size() > 0;
-            if (playing) {
-                JsonObject player0 = response.getAsJsonArray().get(0).getAsJsonObject();
-                return player0.get("playerid").getAsInt();
+
+        if (response instanceof JsonArray) {
+            JsonArray result = response.getAsJsonArray();
+            if (result.size() > 0) {
+                JsonObject player0 = result.get(0).getAsJsonObject();
+                if (player0.has("playerid")) {
+                    return player0.get("playerid").getAsInt();
+                }
             }
         }
         return -1;
@@ -81,6 +84,7 @@ public class KodiConnection implements KodiClientSocketEventListener {
 
     public synchronized void playerPlayPause() {
         int activePlayer = getActivePlayer();
+
         JsonObject params = new JsonObject();
         params.addProperty("playerid", activePlayer);
         socket.callMethod("Player.PlayPause", params);
@@ -88,6 +92,7 @@ public class KodiConnection implements KodiClientSocketEventListener {
 
     public synchronized void playerStop() {
         int activePlayer = getActivePlayer();
+
         JsonObject params = new JsonObject();
         params.addProperty("playerid", activePlayer);
         socket.callMethod("Player.Stop", params);
@@ -176,8 +181,13 @@ public class KodiConnection implements KodiClientSocketEventListener {
         params.add("properties", getJsonArray(properties));
         JsonElement response = socket.callMethod("Player.GetProperties", params);
 
-        JsonObject result = response.getAsJsonObject();
-        return result.get("speed").getAsInt();
+        if (response instanceof JsonObject) {
+            JsonObject result = response.getAsJsonObject();
+            if (result.has("speed")) {
+                return result.get("speed").getAsInt();
+            }
+        }
+        return 0;
     }
 
     public synchronized void updatePlayerStatus() {
@@ -231,58 +241,63 @@ public class KodiConnection implements KodiClientSocketEventListener {
         params.add("properties", getJsonArray(properties));
         JsonElement response = socket.callMethod("Player.GetItem", params);
 
-        JsonObject item = ((JsonObject) response).get("item").getAsJsonObject();
+        if (response instanceof JsonObject) {
+            JsonObject result = response.getAsJsonObject();
+            if (result.has("item")) {
+                JsonObject item = result.get("item").getAsJsonObject();
 
-        String title = "";
-        if (item.has("title")) {
-            title = convertToText(item.get("title"));
-        }
+                String title = "";
+                if (item.has("title")) {
+                    title = convertToText(item.get("title"));
+                }
 
-        String showTitle = "";
-        if (item.has("showtitle")) {
-            showTitle = convertToText(item.get("showtitle"));
-        }
+                String showTitle = "";
+                if (item.has("showtitle")) {
+                    showTitle = convertToText(item.get("showtitle"));
+                }
 
-        String album = "";
-        if (item.has("album")) {
-            album = convertToText(item.get("album"));
-        }
+                String album = "";
+                if (item.has("album")) {
+                    album = convertToText(item.get("album"));
+                }
 
-        String mediaType = item.get("type").getAsString();
-        if ("channel".equals(mediaType) && item.has("channeltype")) {
-            String channelType = item.get("channeltype").getAsString();
-            if ("radio".equals(channelType)) {
-                mediaType = "radio";
+                String mediaType = item.get("type").getAsString();
+                if ("channel".equals(mediaType) && item.has("channeltype")) {
+                    String channelType = item.get("channeltype").getAsString();
+                    if ("radio".equals(channelType)) {
+                        mediaType = "radio";
+                    }
+                }
+
+                String artist = "";
+                if ("movie".equals(mediaType)) {
+                    artist = convertFromArray(item.get("director").getAsJsonArray());
+                } else {
+                    if (item.has("artist")) {
+                        artist = convertFromArray(item.get("artist").getAsJsonArray());
+                    }
+                }
+
+                String channel = "";
+                if (item.has("channel")) {
+                    channel = item.get("channel").getAsString();
+                }
+
+                try {
+                    listener.updateAlbum(album);
+                    listener.updateTitle(title);
+                    listener.updateShowTitle(showTitle);
+                    listener.updateArtist(artist);
+                    listener.updateMediaType(mediaType);
+                    listener.updatePVRChannel(channel);
+                } catch (Exception e) {
+                    logger.error("Event listener invoking error", e);
+                }
+
+                if (item.has("fanart")) {
+                    updateFanartUrl(item.get("fanart").getAsString());
+                }
             }
-        }
-
-        String artist = "";
-        if ("movie".equals(mediaType)) {
-            artist = convertFromArray(item.get("director").getAsJsonArray());
-        } else {
-            if (item.has("artist")) {
-                artist = convertFromArray(item.get("artist").getAsJsonArray());
-            }
-        }
-
-        String channel = "";
-        if (item.has("channel")) {
-            channel = item.get("channel").getAsString();
-        }
-
-        try {
-            listener.updateAlbum(album);
-            listener.updateTitle(title);
-            listener.updateShowTitle(showTitle);
-            listener.updateArtist(artist);
-            listener.updateMediaType(mediaType);
-            listener.updatePVRChannel(channel);
-        } catch (Exception e) {
-            logger.error("Event listener invoking error", e);
-        }
-
-        if (item.has("fanart")) {
-            updateFanartUrl(item.get("fanart").getAsString());
         }
     }
 
@@ -466,11 +481,10 @@ public class KodiConnection implements KodiClientSocketEventListener {
 
             JsonObject params = new JsonObject();
             params.add("properties", getJsonArray(props));
-
             JsonElement response = socket.callMethod("Application.GetProperties", params);
 
             if (response instanceof JsonObject) {
-                JsonObject result = (JsonObject) response;
+                JsonObject result = response.getAsJsonObject();
                 if (result.has("volume")) {
                     volume = result.get("volume").getAsInt();
                     listener.updateVolume(volume);
@@ -479,7 +493,6 @@ public class KodiConnection implements KodiClientSocketEventListener {
                     boolean muted = result.get("muted").getAsBoolean();
                     listener.updateMuted(muted);
                 }
-
             }
         } else {
             listener.updateMuted(false);
@@ -501,14 +514,13 @@ public class KodiConnection implements KodiClientSocketEventListener {
         params.addProperty("channeltype", channelType);
         JsonElement response = socket.callMethod("PVR.GetChannelGroups", params);
 
-        JsonArray channelGroups = null;
         if (response instanceof JsonObject) {
-            JsonObject result = (JsonObject) response;
+            JsonObject result = response.getAsJsonObject();
             if (result.has("channelgroups")) {
-                channelGroups = result.get("channelgroups").getAsJsonArray();
+                return result.get("channelgroups").getAsJsonArray();
             }
         }
-        return channelGroups;
+        return null;
     }
 
     public int getChannelGroupID(final String channelType, final String channelGroupName) {
@@ -530,14 +542,13 @@ public class KodiConnection implements KodiClientSocketEventListener {
         params.addProperty("channelgroupid", channelGroupID);
         JsonElement response = socket.callMethod("PVR.GetChannels", params);
 
-        JsonArray channels = null;
         if (response instanceof JsonObject) {
-            JsonObject result = (JsonObject) response;
+            JsonObject result = response.getAsJsonObject();
             if (result.has("channels")) {
-                channels = result.get("channels").getAsJsonArray();
+                return result.get("channels").getAsJsonArray();
             }
         }
-        return channels;
+        return null;
     }
 
     public int getChannelID(final int channelGroupID, final String channelName) {
@@ -601,10 +612,10 @@ public class KodiConnection implements KodiClientSocketEventListener {
 
             JsonObject params = new JsonObject();
             params.add("properties", getJsonArray(props));
-
             JsonElement response = socket.callMethod("Application.GetProperties", params);
+
             if (response instanceof JsonObject) {
-                JsonObject result = (JsonObject) response;
+                JsonObject result = response.getAsJsonObject();
                 if (result.has("version")) {
                     JsonObject version = result.get("version").getAsJsonObject();
                     int major = version.get("major").getAsInt();

